app/src/main/java/tech/relaycorp/letro/account/storage/AccountRepository.kt

```diff
@@ -9,13 +9,26 @@ import kotlinx.coroutines.flow.MutableStateFlow
 import kotlinx.coroutines.launch
 import tech.relaycorp.letro.account.model.Account
 import tech.relaycorp.letro.main.MainViewModel
+import tech.relaycorp.letro.utils.i18n.normaliseString
+import java.security.PrivateKey
+import java.util.Locale
 import javax.inject.Inject
 
 interface AccountRepository {
     val currentAccount: Flow<Account?>
-    suspend fun createAccount(id: String)
+    suspend fun createAccount(
+        requestedUserName: String,
+        domainName: String,
+        locale: Locale,
+        veraidPrivateKey: PrivateKey,
+    )
 
-    suspend fun updateAccountId(id: String, newId: String)
+    suspend fun getByRequest(
+        requestedUserName: String,
+        locale: Locale,
+    ): Account?
+
+    suspend fun completeRegistration(id: Long, veraidId: String, veraidBundle: ByteArray)
```

Comment:
Let's name it `updateAccount()` to not be confused that we have some registration logic in several repositories (`RegistrationRepository` and `AccountRepository`). In fact, here we're updating the account data, so `updateAccount()` seems as an appropriate naming for me. WDYT?